ddtrace/internal/_queue.pyx

```diff
@@ -0,0 +1,52 @@
+import threading
+
+from ddtrace.vendor import attr
+from ..internal.logger import get_logger
+from . import _rand
+
+
+log = get_logger(__name__)
+
+
+@attr.s
+class TraceQueue(object):
+
+    maxsize = attr.ib(type=int, default=0)
+    _lock = attr.ib(init=False, factory=threading.Lock, repr=False)
+    _queue = attr.ib(init=False, factory=list, repr=False)
+    _accepted = attr.ib(init=False, type=int, default=0)
+    _accepted_lengths = attr.ib(init=False, type=int, default=0)
+    _dropped = attr.ib(init=False, type=int, default=0)
+
+    def __len__(self):
+        return len(self._queue)
+
+    def put(self, item):
+        with self._lock:
+            if self.maxsize <= 0 or len(self._queue) < self.maxsize:
+                self._queue.append(item)
+            else:
+                idx = _rand.rand64bits() % len(self._queue)
+                self._queue[idx] = item
+
+                self._dropped += 1
+                log.warning("Trace queue %r is full, dropping a random trace", self)
```

Comment:
```suggestion
                log.warning("%r is full, dropping a random trace", self)
```

Otherwise it'll just say `Trace queue TraceQueue()`, which is mostly redundant.